### PR TITLE
feat(core): Introduce AWS secrets manager as external secrets store

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/PAY-1481

### CODE REVIEW

File: packages/editor-ui/src/components/ExternalSecretsProviderCard.ee.vue [hunk 1]
```diff
@@ -54,7 +54,7 @@ const canConnect = computed(() => {
 });
 
 const formattedDate = computed((provider: ExternalSecretsProvider) => {
-	return DateTime.fromISO(props.provider.connectedAt).toFormat('dd LLL yyyy');
+	return DateTime.fromISO(props.provider.connectedAt ?? new Date()).toFormat('dd LLL yyyy');
```
Comment (ivov) [inline packages/editor-ui/src/components/ExternalSecretsProviderCard.ee.vue:57]: Fix for pre-existing bug where `props.provider.connectedAt` was `null` immediately on connect, causing `Invalid DateTime` to display.

File: packages/cli/src/Interfaces.ts [hunk 2]
```diff
@@ -669,7 +669,7 @@ export abstract class SecretsProvider {
 	abstract disconnect(): Promise<void>;
 	abstract update(): Promise<void>;
 	abstract test(): Promise<[boolean] | [boolean, string]>;
-	abstract getSecret(name: string): IDataObject | undefined;
+	abstract getSecret(name: string): unknown;
```
Comment (ivov) [inline packages/cli/src/Interfaces.ts:672]: See [here](https://n8nio.slack.com/archives/C04B1GZ4T0U/p1711524858029009).

File: packages/cli/src/ExternalSecrets/providers/aws-secrets/aws-secrets-manager.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+	getSecretNames() {
+		return Object.keys(this.cachedSecrets);
+	}
+
+	private assertAuthType(context: AwsSecretsManagerContext) {
+		if (context.settings.authMethod === 'iamUser') return;
+
+		throw new UnknownAuthTypeError(context.settings.authMethod);
+	}
```
Comment (valya) [inline packages/cli/src/ExternalSecrets/providers/aws-secrets/aws-secrets-manager.ts:125]: Do we have any plans to include using instance permissions (e.g. whatever permissions an EC2 instance happens to have)?

**Review by valya — DISMISSED** [review]

File: packages/cli/src/ExternalSecrets/providers/aws-secrets/aws-secrets-manager.ts (hunk 3 above)
Comment (ivov) [reply packages/cli/src/ExternalSecrets/providers/aws-secrets/aws-secrets-manager.ts:125]: Aiming to add more auth types in future, but we haven't discussed details yet.

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.36.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.36.0)

**Pull request merged**